server/src/WebSocketEvents.js

```diff
@@ -763,6 +763,20 @@ eventResponses.start_dungeon = (clientSocket, data) => {
     dungeonPortal.enter(player);
 };
 
+/**
+ * @param {*} clientSocket
+ * @param {String} data.dungeonID - The ID of a dungeon manager.
+ */
+ eventResponses.leave_dungeon = (clientSocket) => {
+    if (clientSocket.inGame === false) return;
+
+    /** @type {Dungeon} */
+    const dungeon = clientSocket.entity.board.dungeon;
```

Comment:
A check for whether the board property of the entity isn't null is needed before here, as a dead player in the dungeon would be able to send the "leave_dungeon" event, but when they die their `board` property is nulled, so accessing `dungeon` in `clientSocket.entity.board.dungeon` would be undefined and throw an exception.

This effect can be produced by going into a dungeon, enter `/killself` in chat, then in the browser console, enter `dungeonz.states.ApplicationState.connection.sendEvent("leave_dungeon");` to manually send the leave dungeon event.

Reply:
Would the appropriate behaviour in this case be to return and do nothing if the board is null, or would the event handler need to find some way to spawn the player in the overworld?

Reply:
It is fine to just return like this.
It is as edge-case problem that can only be triggered by a user manually sending this event (i.e. through the console), as when their character is in the dead state, the client should only show them the respawn option on the GUI, so they shouldn't normally be able to press the leave dungeon button while dead.